lib/Service/ApiService.php

```diff
@@ -67,4 +81,35 @@ public function sendFile(int $fileId): void {
 			],
 		);
 	}
+
+	public function searchMessages(string $userId, string $term, int $offset = 0, int $limit = 10): array {
+		$arguments = [
+			'format' => 'json',
+			'query' => '*' . $term . '*' ,
+		];
+
+		$paperlessURL = rtrim($this->config->url, '/') . '/api/documents/?' . http_build_query($arguments);
+		$result = $this->client->get($paperlessURL,
+			[
+				'headers' => array_merge(
+					$this->getAuthorizationHeaders(),
+					[
+						'Content-Type' => 'application/json',
+						'Accept' => 'application/json'
+					]
+				)
+			]
+		);
+
+		$body = $result->getBody();
+		$json_body = json_decode($body, true);
```

Comment:
```suggestion
		$json_body = json_decode($body, true, 512, JSON_THROW_ON_ERROR);
```
To ensure any decoding error will trigger an error.